source/geometries/Next100Vessel.cc

```diff
@@ -253,86 +311,57 @@ namespace nexus {
-      vacuum_manifold_gas_logic->SetVisAttributes(G4VisAttributes::Invisible);
     }
 
-
     // VERTEX GENERATORS   //////////
-    body_gen_  = new CylinderPointSampler(vessel_in_rad_, vessel_length_, vessel_thickness_, 0.);
+    body_gen_  = new CylinderPointSampler(vessel_in_rad_, body_length_, vessel_thickness_, 0.);
 
-    tracking_endcap_gen_ = new SpherePointSampler( endcap_in_rad_, endcap_thickness_, tracking_endcap_pos, 0,
+    energy_endcap_gen_ = new SpherePointSampler( endcap_in_rad_, vessel_thickness_, energy_endcap_pos, 0,
 						   0., twopi, 0., endcap_theta_);
 
-    energy_endcap_gen_ = new SpherePointSampler( endcap_in_rad_, endcap_thickness_, energy_endcap_pos, 0,
-						 0., twopi, 180.*deg - endcap_theta_, endcap_theta_);
+    tracking_endcap_gen_ = new SpherePointSampler( endcap_in_rad_, vessel_thickness_, tracking_endcap_pos, xRot,
+						 0., twopi, 0., endcap_theta_);
+
+    tracking_flange_gen_  = new CylinderPointSampler(vessel_out_rad, flange_tp_length,
+						     flange_out_rad-vessel_out_rad, 0., tracking_flange_pos);
 
-    tracking_flange_gen_  = new CylinderPointSampler(vessel_out_rad, flange_length_,
-						     flange_out_rad_-vessel_out_rad, 0., tracking_flange_pos);
+    energy_flange_gen_  = new CylinderPointSampler(vessel_out_rad, flange_ep_length,
+						   flange_out_rad-vessel_out_rad, 0., energy_flange_pos);
 
-    energy_flange_gen_  = new CylinderPointSampler(vessel_out_rad, flange_length_,
-						   flange_out_rad_-vessel_out_rad, 0., energy_flange_pos);
+    port_gen_ = new CylinderPointSampler(0., (port_base_height+vessel_thickness_), port_in_rad, 0.);
 
     // Calculating some prob
-    G4double body_vol = vessel_body_solid->GetCubicVolume() - vessel_gas_body_solid->GetCubicVolume();
-    G4double endcap_vol =  vessel_tracking_endcap_solid->GetCubicVolume() - vessel_gas_tracking_endcap_solid->GetCubicVolume();
-    perc_endcap_vol_ = endcap_vol / (body_vol + 2. * endcap_vol);
+    G4double body_vol   = vessel_body_solid  ->GetCubicVolume() - vessel_gas_body_solid  ->GetCubicVolume();
+    G4double endcap_vol = vessel_endcap_solid->GetCubicVolume() - vessel_gas_endcap_solid->GetCubicVolume();
+    G4double flange_ep_vol = vessel_ep_flange_solid->GetCubicVolume();
+    G4double flange_tp_vol = vessel_tp_flange_solid->GetCubicVolume();
+    G4double vessel_vol = body_vol + 2.*endcap_vol + flange_ep_vol + flange_tp_vol;
 
-  }
+    perc_endcap_vol_    = 2.*endcap_vol / vessel_vol;
+    perc_ep_flange_vol_ = flange_ep_vol / vessel_vol;
+    perc_tp_flange_vol_ = flange_tp_vol / vessel_vol;
 
+    // G4double body_vol = vessel_body_solid->GetCubicVolume() - vessel_gas_body_solid->GetCubicVolume();
+    // std::cout << "Body vol (cm3)" << body_vol/1e6 << '\n';
+    // G4double endcap_vol = vessel_endcap_solid->GetCubicVolume() - vessel_gas_endcap_solid->GetCubicVolume();
+    // std::cout << "Endcap vol (cm3)" << endcap_vol/1e6 << '\n';
```

Comment:
Why are these lines commented?

Reply:
Yes, could you remove them? I think they are not necessary.

Reply:
Done